src/pages/Store/StorePage/components/EventCarousel/index.tsx

```diff
@@ -65,112 +53,119 @@ function Card({
         </div>
         <div className={styles['swipe-item__nowrap']}>할인 혜택을 받아보세요!</div>
       </div>
-    </Link>
+    </button>
   );
 }
 
-function EventCarousel() {
-  const carouselList = useGetAllEvents();
+export default function EventCarousel() {
   const isMobile = useMediaQuery();
-  const {
-    emblaRef, canNextClick, canPrevClick, currentIndex, scrollTo,
-  } = useCarouselController(isMobile);
   const logger = useLogger();
-  const eventLogging = (shopName: string) => {
+  const { events } = useGetAllEvents();
+  const { emblaRef, currentIndex, scrollTo } = useCarouselController(isMobile);
+
+  const eventLogging = (shopId: number, shopName: string) => {
+    if (shopId === 0) return;
+
     logger.actionEventClick({
       actionTitle: 'BUSINESS', event_label: 'shop_categories_event', value: `${shopName}`, event_category: 'click',
     });
   };
 
-  if (!carouselList) return null;
-  if (carouselList.length < 1) return null;
+  if (events.length < 1) return null;
+
+  const getEditedEvents = () => {
+    if (isMobile) return events.slice(0, 10);
+    if (events.length % 2 === 0) return events;
+    return ([...events, {
+      shop_id: 0,
+      event_id: 0,
+      shop_name: '코인',
+      thumbnail_images: ['http://static.koreatech.in/assets/img/rectangle_icon.png'],
+    }]);
+  };
+
+  const getCurrentIndex = () => {
+    const currentPage = currentIndex + 1;
+    const denominator = Math.min(10, events.length);
+    return `${currentPage}/${denominator}`;
+  };
 
   if (isMobile) {
     return (
-      <div className={styles.carousel}>
-        <div className={styles.container} ref={emblaRef}>
-          <div className={styles.swipe}>
-            {carouselList.slice(0, 10).map((item) => (
-              <Card
-                shop_id={String(item.shop_id)}
-                event_id={item.event_id}
-                shop_name={item.shop_name}
-                thumbnail_images={item.thumbnail_images}
-                onClick={() => eventLogging(item.shop_name)}
-              />
-            ))}
-          </div>
-          <div className={styles.pagination}>
-            { // eslint-disable-next-line
+      <Suspense fallback={null}>
+        <div className={styles.carousel}>
+          <div className={styles.container} ref={emblaRef}>
+            <div className={styles.swipe}>
+              {getEditedEvents().map((item) => (
```

Comment:
```suggestion
              const editedEvents = getEditedEvents();
// ==========
              {editedEvents.map((item) => (
```

변수에 할당하고 사용하는 건 어떤가요?
변수에 할당을 하게 된다면 가독성이나 디버깅 할 때도 값을 console 찍어보기 더 편할 것 같아요

Reply:
즉시 실행 함수로 처음부터 만들면서 할당 할 수 있었군요!